CMSgov/design-system
# [WNMGDS-1589] gatsby mobile nav

## Summary

Adding styling for the mobile nav for Gatsby doc site.

### Added

* Mobile menu button

### Changed

* Renaming `DocSiteNav` to `DocSiteSidebar`
* Made a `GithubLinks` component for reuse in desktop & mobile 

## How to test

`yarn install` & `yarn start:gatsby`. Change the screen size and confirm that nav switches between mobile and desktop views
![DocSiteMobileMenu](https://user-images.githubusercontent.com/33579665/164499325-b6502e31-9283-4e01-be8c-a79c9fc24e54.gif)
